worldedit-cli/src/main/java/com/sk89q/worldedit/cli/CLIWorldEdit.java

```diff
@@ -0,0 +1,332 @@
+    }
+
+    public void onStopped() {
+        WorldEdit worldEdit = WorldEdit.getInstance();
+        worldEdit.getSessionManager().unload();
+        worldEdit.getPlatformManager().unregister(platform);
+    }
+
+    public FileRegistries getFileRegistries() {
+        return this.fileRegistries;
+    }
+
+    /**
+     * Get the configuration.
+     *
+     * @return the CLI configuration
+     */
+    CLIConfiguration getConfig() {
+        return this.config;
+    }
+
+    /**
+     * Get the WorldEdit proxy for the platform.
+     *
+     * @return the WorldEdit platform
+     */
+    public Platform getPlatform() {
+        return this.platform;
+    }
+
+    /**
+     * Get the working directory where WorldEdit's files are stored.
+     *
+     * @return the working directory
+     */
+    public Path getWorkingDir() {
+        return this.workingDir;
+    }
+
+    /**
+     * Get the version of the WorldEdit-CLI implementation.
+     *
+     * @return a version string
+     */
+    String getInternalVersion() {
+        if (version == null) {
+            version = getClass().getPackage().getImplementationVersion();
+        }
+        return version;
+    }
+
+    public void saveAllWorlds(boolean force) {
+        platform.getWorlds().stream()
+                .filter(world -> world instanceof CLIWorld)
+                .forEach(world -> ((CLIWorld) world).save(force));
+    }
+
+    public void run(InputStream inputStream) {
+        try (Scanner scanner = new Scanner(inputStream)) {
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                if (line.equals("stop")) {
+                    commandSender.print("Stopping!");
+                    break;
+                }
+                CommandEvent event = new CommandEvent(commandSender, line);
+                WorldEdit.getInstance().getEventBus().post(event);
+                if (!event.isCancelled()) {
+                    commandSender.printError("Unknown command!");
+                } else {
+                    saveAllWorlds(false);
+                }
+            }
+        } finally {
+            saveAllWorlds(false);
+        }
+    }
+
+    public static void main(String[] args) {
+        Options options = new Options();
+        options.addOption("f", "file", true, "The file to load in. Either a schematic, or a level.dat in a world folder.");
+        options.addOption("s", "script", true, "A file containing a list of commands to run. Newline separated.");
+        int exitCode = 0;
+
+        CLIWorldEdit app = new CLIWorldEdit();
+        app.onInitialized();
+
+        InputStream inputStream = System.in;
+
+        try {
+            CommandLine cmd = new DefaultParser().parse(options, args);
+
+            String fileArg = cmd.getOptionValue('f');
+            File file;
+            if (fileArg == null) {
+                file = app.commandSender.openFileOpenDialog(new String[]{"schem", "dat"});
```

Comment:
Should be dynamically built from formats now.

Reply:
Oh, true